torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -431,26 +433,40 @@ def powerSGD_hook(
             total_Qs_size, device=device, dtype=dtype
         )
 
+    # Batch tensors to compress by shape.
+    shape_to_tensors: Dict[torch.Size, List[torch.Tensor]] = {}
```

Comment:
Optional: you can just use `shape_to_tensors = defaultdict(list)` to save L439-L440.